libkbfs/mdserver_memory.go

```diff
@@ -371,8 +371,11 @@ func (md *MDServerMemory) Put(ctx context.Context, rmds *RootMetadataSigned) err
 
 	// Consistency checks
 	if head != nil {
-		err := head.MD.CheckValidSuccessorForServer(
-			md.config.Crypto(), &rmds.MD)
+		id, err := md.config.Crypto().MakeMdID(&head.MD)
```

Comment:
I'm a little worried about the new extra calls to `MakeMdId`, since in most local-server KBFS profiles, hashing is the biggest CPU user, and not re-using the value calculated by the client will result in even more hashing.  That said, I understand why you did it this way, and I don't have a concrete suggestion on how to make it better.  Just something to keep in mind; plus it might be worth comparing the test speed before and after this change to make sure it didn't slow things down too much.